index.js

```diff
@@ -104,7 +104,10 @@ var publicAgent = {
 
     common.utils.getHostname(function(err, hostname) {
       if (err) {
-        logger.info('Unable to retrieve GCE hostname.', err);
+        if (err.code !== 'ENOTFOUND') {
+          logger.warn('Unable to retrieve GCE hostname.', err);
+        }
+        config.hostname = require('os').hostname();
```

Comment:
I think it would be best if we made the functions inside util just connect us to different pieces of metadata from the metadata service and resolve local options inside the trace agent the same way we do for project number currently.